### PR TITLE
Add test:e2e target

### PR DESCRIPTION
**Pull request by thibmeu**
Allow testing an issuer end-to-end from the command line. Locally, the client fetches the issuer public key and token request uri. It then generates a token request, sends it, and validates the received token.

mTLS is supported, as it's a popular authentication method.

### CODE REVIEW

File: test/e2e/index.ts [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	const tokenRequest = await client.createTokenRequest(challenge, issuerPublicKeyEnc);
+
+	const proxyFetch = mTLS ? await fetchWithMTLS(mTLS) : fetch;
+	const response = await proxyFetch(issuerRequestURL, {
+		method: 'POST',
+		headers: {
+			'Content-Type': MediaType.PRIVATE_TOKEN_REQUEST,
+			'Accept': MediaType.PRIVATE_TOKEN_RESPONSE,
+		},
+		body: Buffer.from(tokenRequest.serialize()),
```
Comment (armfazh) [inline test/e2e/index.ts:103]: Can we remove Buffer?
```suggestion
		body: tokenRequest.serialize().buffer,
```

File: test/e2e/index.ts [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+	console.log(
+		`Issuer tokens are valid: ${
+			(await verifyToken(BlindRSAMode.PSS, token, issuerPublicKey)) &&
+			response.headers.get('Content-Type') === MediaType.PRIVATE_TOKEN_RESPONSE
+		}`
+	);
+}
+
+async function main() {
```
Comment (armfazh) [inline test/e2e/index.ts:123]: the entire body of the test seems to me that can be wrapped with a regular Jest test.
No need of using a CLI, other that for passing params, which can be passed and stored in a `testdata` folder.

**Review by armfazh — CHANGES_REQUESTED** [review]

File: test/e2e/index.ts (hunk 1 above)
Comment (thibmeu) [reply test/e2e/index.ts]: I've updated the type to use .buffer and cast `Buffer`. This is a [type error](https://github.com/node-fetch/node-fetch/pull/1521) on `node-fetch` side.

File: test/e2e/index.ts (hunk 2 above)
Comment (thibmeu) [reply test/e2e/index.ts:123]: I agree. At the same time, this repository is not tighten to any specific deployment.
I'll see how to update this repository to use wrangler.toml configuration in test:e2e target.

**Comment by thibmeu** [comment]
@armfazh I've kept the command as an independant process, but have also added an e2e test to jest, so it runs automatically on localhost.

File: package.json [hunk 3]
```diff
@@ -10,7 +10,8 @@
 		"dev": "wrangler dev --env production --test-scheduled",
 		"format": "prettier --write  '*.{json,js}' 'scripts/*.{js,ts}' 'src/**/*.{js,ts}' 'test/**/*.{js,ts}'",
 		"lint": "eslint src && prettier --check '*.{json,js}' 'scripts/*.{js,ts}' 'src/**/*.{js,ts}' 'test/**/*.{js,ts}'",
-		"test": "npm run build && node --experimental-vm-modules --no-warnings node_modules/.bin/jest"
+		"test": "npm run build && node --experimental-vm-modules --no-warnings node_modules/.bin/jest",
+		"test:e2e": "tsc -p ./test/e2e/tsconfig.json && node --experimental-vm-modules --no-warnings dist/index.js"
```
[the remark this answers is not in the document]
Comment (armfazh) [reply package.json:14]: See my proposed changes at https://github.com/thibmeu/pp-issuer/pull/1

**Review by armfazh — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (thibmeu) [reply package.json]: the proposal is great. i've merged it into this branch

**Pull request merged**